src/utils/__tests__/style-formatter.spec.js

```diff
@@ -31,133 +34,145 @@ describe("style-formatter", () => {
     });
 
     it("should return specified background color from expression", () => {
       style.background.colorExpression = someColorExpression;
       style.background.useColorExpression = true;
-      const formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app });
+      const formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app, imageUrl });
       expect(formattedStyle.includes(`background-color: ${someColorExpression}`)).toBe(true);
     });
 
     it("should return default background color when color is none", () => {
-      const formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app });
+      const formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app, imageUrl });
       expect(formattedStyle.includes("background-color: myPrimaryColor")).toBe(true);
     });
 
-    it("should return specified image url when the background mode is set on media", () => {
+    it("should set the image url when the image url exists", async () => {
       style.background.useImage = false;
       style.background.mode = "media";
       style.background.url.qStaticContentUrl = { qUrl: someUrl };
-      const formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app });
-      expect(formattedStyle.includes(`background-image: url('https://localhost${someUrl}')`)).toBe(true);
+      imageUrl = await new Promise((resolve) => {
+        setTimeout(() => resolve(`https://localhost${someUrl}`), 1000);
+      });
+      await waitFor(() => expect(imageUrl).toBe(`https://localhost${someUrl}`));
+      const formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app, imageUrl });
+      expect(formattedStyle.includes(`background-image: url('${imageUrl}')`)).toBe(true);
       expect(formattedStyle.includes("background-size: auto auto")).toBe(true);
       expect(formattedStyle.includes("background-position: 50% 50%")).toBe(true);
       expect(formattedStyle.includes("background-repeat: no-repeat")).toBe(true);
     });
 
-    it("should not show size/position when background mode is set on media but the url is not set", () => {
+    it("should not set the image url when the image is not loaded", async () => {
       style.background.useImage = false;
       style.background.mode = "media";
-      style.background.url.qStaticContentUrl = {};
-      const formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app });
-      expect(formattedStyle.includes("background-image:")).toBe(false);
-      expect(formattedStyle.includes("background-size:")).toBe(false);
-      expect(formattedStyle.includes("background-position:")).toBe(false);
-      expect(formattedStyle.includes("background-repeat:")).toBe(false);
+      style.background.url.qStaticContentUrl = { qUrl: someUrl };
+      imageUrl = await new Promise((reject) => {
+        setTimeout(() => reject(undefined), 500);
+      });
+      await waitFor(() => expect(imageUrl).toBe(undefined));
```

Comment:
This is just verifying the promise that was created on the previous line.

What is that you really want to test here?